doc/manual/config-basic.rst

```diff
@@ -107,6 +107,149 @@ Problems can have special *compare* and
 *run* scripts associated to them, to deal with problem
 statements that require non-standard evaluation.
 
+Adding contest data programmatically
+------------------------------------
+
+DOMjudge offers API endpoints to add or update contest data programmatically.
+In general, we try to follow the `CLICS CCS specification
+<https://clics.ecs.baylor.edu/index.php?title=Contest_Control_System_Requirements>`_
+for all file formats involved.
+
+All of the following examples require you to set up admin credentials in your
+`.netrc <https://www.gnu.org/software/inetutils/manual/html_node/The-_002enetrc-file.html>`_ file.
+You need to install `httpie <https://httpie.org/>`_ and replace the
+``<API_URL>`` in the examples below with the API URL of your local DOMjudge
+installation.
+
+Importing team categories
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Prepare a file called ``groups.tsv`` which contains the team categories.
+The first line should contain ``File_Version 1``.
+Each of the following lines must contain the following elements separated by tabs:
+
+- the category ID
+- the name of the team category
+
+Example ``groups.tsv``::
+
+   File_Version   1
+   13337	Companies
+   47	Netherlands
+   23	United Kingdom
+
+To import the file run the following command::
+
+    http --check-status -b -f POST "<API_URL>/users/groups" tsv@groups.tsv
+
+Importing teams
+^^^^^^^^^^^^^^^
+
+Prepare a file called ``teams2.tsv`` which contains the teams.
+The first line should contain ``File_Version	2``.
+Each of the following lines must contain the following elements separated by tabs:
+
+- the team ID
+- an external ID, e.g. from the ICPC CMS, may be empty
+- the category ID
+- the team name
+- the institution name
+- the institution short name
+- a country code in form of ISO 3166-1 alpha-3
+- an external institution ID, e.g. from the ICPC CMS, may be empty
+
+Example ``teams2.tsv``::
```

Comment:
General note: should we also add JSON examples, since that is the new way of doing teams, affiliations and categories?

Reply:
We could do in a separate commit. This already provides value and adds things we have tested at NWERC.

I also think these file formats should be updated first: https://clics.ecs.baylor.edu/index.php?title=Contest_Control_System_Requirements#Appendix:_File_formats
Perhaps this is a good topic for your CLICS calls.